src/app/admin/components/microschema-detail/mircoschema-detail.component.ts

```diff
@@ -56,10 +84,49 @@ export class MicroschemaDetailComponent implements OnInit, OnDestroy {
                 : `{}`;
             this.ref.detectChanges();
         });
+
+        this.microschema$
+            .take(1)
+            .toPromise()
+            .then(microschema => this.schemaEffects.loadEntityAssignments('microschema', microschema.uuid))
+            .then(assignments => (this.projectAssignments = assignments));
+
+        this.adminProjectEffects.loadProjects();
+
+        this.filterInput.valueChanges
+            .debounceTime(100)
+            .takeUntil(this.destroy$)
+            .subscribe(term => {
+                setQueryParams(this.router, this.route, { q: term });
+            });
+
+        observeQueryParam(this.route.queryParamMap, 'q', '')
+            .takeUntil(this.destroy$)
+            .subscribe(filterTerm => {
+                this.adminProjectEffects.setFilterTerm(filterTerm);
+                this.filterInput.setValue(filterTerm, { emitEvent: false });
+            });
+
+        const allProjects$ = this.state
+            .select(state => state.adminProjects.projectList)
+            .map(uuids => uuids.map(uuid => this.entities.getProject(uuid)).filter(notNullOrUndefined));
+
+        const filterTerm$ = this.state.select(state => state.adminProjects.filterTerm);
+
+        this.projects$ = combineLatest(allProjects$, filterTerm$).map(([projects, filterTerm]) => {
+            this.filterTerm = filterTerm;
+            return projects.filter(project => fuzzyMatch(filterTerm, project.name) !== null).sort((pro1, pro2) => {
+                return pro1.name < pro2.name ? -1 : 1;
+            });
+        });
     }
 
     ngOnDestroy(): void {
-        this.subscription.unsubscribe();
+        if (this.subscription) {
+            this.subscription.unsubscribe();
+        }
+        this.destroy$.next();
```

Comment:
Check if this is necessary

Reply:
It is not necessary, was removed from:
- mircoschema-detail.component.ts;
- schema-detail.component.ts;
- mircoschema-list.component.ts;
- mircoschema-list.component.ts.